synapse/storage/databases/main/events.py

```diff
@@ -2192,9 +2192,9 @@ def _set_push_actions_for_event_and_users_txn(
         sql = """
             INSERT INTO event_push_actions (
                 room_id, event_id, user_id, actions, stream_ordering,
-                topological_ordering, notif, highlight, unread
+                topological_ordering, notif, highlight, unread, thread_id
             )
-            SELECT ?, event_id, user_id, actions, ?, ?, notif, highlight, unread
+            SELECT ?, event_id, user_id, actions, ?, ?, notif, highlight, unread, thread_id
```

Comment:
Maybe we should be providing the `thread_id` like we do the `room_id`? (I.e. not storing it in `event_push_actions_staging` and only storing it in `event_push_actions`)